### ISSUE TITLE
Calculate Nonce

### ISSUE THREAD

**Issue by JJTech0130** [Leseratte10/acsm-calibre-plugin#27]
I'm implementing this in Lua (for real this time :) so that I can make a KOReader plugin.
I've gotten through signing in, and now I need to do activation.
The one thing that's tripping me up is calculating this nonce:
```python3
dt = datetime.utcnow()
usec = dt.microsecond
sec = (dt - datetime(1970,1,1)).total_seconds()


Ntime = int(int(sec * 1000) + usec/1000)
```
I think I get the part about converting it to Gregorian after that, and it *looks* like you're just calculating the current unix timestamp, right?
I don't get the whole `usec` thing though... why are you converting the unix timestamp to μs, and then adding the current μs divided by 1000 to it?

**Comment by Leseratte10** [comment]
The end result (the `Ntime`) is supposed to be milliseconds since 1970. However, the datetime object only supports `total_seconds`, not something like `total_milliseconds`. 

So I multiply the seconds by 1000 to get milliseconds, but then I need to add the remaining 0-999 milliseconds from within the current second, and I get these by taking the current microsecond counter (from 0 to 999999) and dividing that by thousand. 

Basically, all these four lines of code are doing is getting the amount of milliseconds since 1970.

**Comment by JJTech0130** [comment]
Ah, that makes sense. But isn't `total_seconds` a decimal? When I put it into an interpreter I get `1657554769.363188`
Part of my confusion was that I mixed up micro and milli seconds a bit :)
